pkg/metrics/statsd/statsd.go

```diff
@@ -0,0 +1,216 @@
+package statsd
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/metrics"
+)
+
+// statsd implementation for https://genevamondocs.azurewebsites.net/collect/references/statsdref.html
+
+// Now stubs time.Now
+var Now = time.Now
+
+var (
+	defaultPrefix = "metrics"
+)
+
+// config defines internal statsd client
+// It should be cloned, but not modified
+type config struct {
+	socket string
+
+	conn net.Conn
+	env  env.Interface
+	mu   sync.Mutex
+}
+
+// clientConfig defines external client config
+type clientConfig struct {
+	metrics.ClientConfig
+	*config
+}
+
+// New method to initialize udp connection
+func New(ctx context.Context, log *logrus.Entry, env env.Interface, opts ...metrics.Option) (metrics.Interface, error) {
+	// defaults
+	cConfig := &metrics.ClientConfig{
+		Account:   os.Getenv("METRICS_ACCOUNT"),
+		Namespace: os.Getenv("METRICS_NAMESPACE"),
+		Dims: map[string]string{
+			"location": env.Location(),
+		},
+		Prefix: defaultPrefix,
+	}
+
+	// apply opts input
+	for _, o := range opts {
+		o(cConfig)
+	}
+
+	// create main config
+	clientConfig := &clientConfig{
+		config: &config{
+			socket: env.MetricsSocket(),
+			env:    env,
+		},
+		ClientConfig: *cConfig,
+	}
+
+	var err error
+	clientConfig.config.conn, err = net.Dial("unix", clientConfig.config.socket)
+	if err != nil && env.IsDevelopment() {
```

Comment:
`_, ok := _env.(env.Dev); ok`
Specifically only ignore ENOENT in development mode.